### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by michalfita** [nuta/kerla#45] [Q&A]
That arose during me writing #44:

# Documentation
There's certain documentation how to do basic stuff with Kerla at its current state, but I don't see any design documents or notes how for example prepare the new target. You started somewhere, would be nice if you share your notes. I know there's great Oppermann's [Writing an OS in Rust](https://os.phil-opp.com/), but I'm interested in something Kerla focused. I'd like to spin Kerla on Raspberry Pi. Any plans regarding better documentation? Or maybe some road map or plans leading people in the right direction when picking up work?

**Comment by nuta** [comment]
I think Kerla needs more documentation especially describing the overview (like [ARCHITECTURE.md](https://matklad.github.io/2021/02/06/ARCHITECTURE.md.html)). I'll work on it this week end 👍

**Comment by nuta** [comment]
BTW, I'm looking forward to seeing ARM (specifically AArch64) support in Kerla too. This is why (almost all) x64-specific logic are isolated into `kernel/arch/x64` :)
